src/index.ts

```diff
@@ -0,0 +1,507 @@
+import * as crypto from 'crypto'
+import * as ethUtil from 'ethereumjs-util'
+
+const bs58check = require('bs58check')
+const randomBytes = require('randombytes')
+const scryptsy = require('scrypt.js')
+const uuidv4 = require('uuid/v4')
+
+// parameters for the toV3() method
+
+interface V3Params {
+  kdf: string
+  cipher: string
+  salt: Buffer
+  iv: Buffer
+  uuid: Buffer
+  dklen: number
+  c: number
+  n: number
+  r: number
+  p: number
+}
+
+function mergeToV3ParamsWithDefaults(params?: Partial<V3Params>): V3Params {
+  const v3Defaults: V3Params = {
+    cipher: 'aes-128-ctr',
+    kdf: 'scrypt',
+    salt: randomBytes(32),
+    iv: randomBytes(16),
+    uuid: randomBytes(16),
+    dklen: 32,
+    c: 262144,
+    n: 262144,
+    r: 8,
+    p: 1,
+  }
+
+  if (!params) {
+    return v3Defaults
+  }
+  return {
```

Comment:
Merging default values with params (also in other functions) can be simplified to something like:

```typescript
return { ...v3Defaults, ...params }
```

Reply:
Agreed